### PR TITLE
Fix/multiple tab containers

### LINKED ISSUE #314: allowing for multiple tabs-containers within a single page
Currently only one is possible. Writing two in the code results in both being combined into one.

### PR DESCRIPTION
**Pull request by broadwell**
This PR is intended to fix #314 (allow multiple tab containers on the same page) and also the Jekyll-related portion of #315 (allow for linking to a specific tab on a page) -- though linking to tabs on pages rendered via React will require a separate fix.
The refinements to the tab-handling code are backwards compatible with the tab markup on existing pages, but also allows for the use of a somewhat simplified tab container template, to be added to the wiki and made known to the PIs when the PR is merged.

### CODE REVIEW

File: src/assets/tabs.js [hunk 1]
```diff
@@ -1,52 +1,92 @@
...
+   * the placement of the tab IDs (in the section body, rather than the link)
+   * otherwise will cause the browser to scroll too far.
+   */
+
+  if (tabID == null && window.location.hash)
+    tabID = window.location.hash.slice(1);
+
+  var containers = document.querySelectorAll("div.tabs-container");
+  containers.forEach(function(container) {
+    var matchedSectionIndex = -1;
```
Comment (simonwiles) [inline src/assets/tabs.js]: If you were looking to optimize for efficiency, you could do all of this without much less looping, e.g.

```
  var section = document.getElementById(tabID);
  section.parentElement.children.forEach(
    function(_section) { _section.style.display = "none"; }
  );
  section.style.display = "block";
```

(Would be even more elegant with arrow functions, but it's probably best not to use them here, as this script is not transpiled,)

File: src/assets/tabs.js [hunk 2]
```diff
@@ -1,52 +1,92 @@
...
-      ).style.display =
-        "block";
+      ul.appendChild(li);
     });
+    container.querySelector("div.wrapper").children[0].appendChild(ul);
   });
-  // Show contents of first tab by default
-  lis[0].click();
+
+  if (window.location.hash) activateTab();
```
Comment (simonwiles) [inline src/assets/tabs.js]: This line would be better at the bottom before attaching the `onHashChange` listener, but it's not likely to matter in this case.

Comment (broadwell) [reply src/assets/tabs.js]: I'm happy to move it there, though in that case I assume the order would be the following?
```
attachTabs();
if (window.location.hash) activateTab();
window.addEventListener("hashchange", activateTab, false);
```

File: src/assets/tabs.js (hunk 1 above)
Comment (broadwell) [reply src/assets/tabs.js]: Some of the looping is left over from when I was trying to do all of this without any element IDs... I'll update with this suggestion.

Comment (broadwell) [reply src/assets/tabs.js]: Huh, `getElementById` seems to fail when the section in question has `display = "none"`. This, combined with the lack of convenient linkages between the tab headers and their content sections (requiring they be matched by their array indices), leads to most of the looping. It's not ideal, but it seems to work well enough, and a major re-implementation might be more work than it's worth, so leaving this for now.

Comment (simonwiles) [reply src/assets/tabs.js]: Agree with leaving it as it is (I've already merged), but surely `getElementById` can't be failing like that...  There must be something else going on.  Multiple (non-unique) IDs, perhaps?

**Pull request merged**